src/Factory/ActivityPub/GroupFactory.php

```diff
@@ -13,13 +15,25 @@ class GroupFactory
 {
     public function __construct(
         private readonly UrlGeneratorInterface $urlGenerator,
+        private readonly MarkdownConverter $markdownConverter,
         private readonly ContextsProvider $contextProvider,
         private readonly ImageManager $imageManager
     ) {
     }
 
     public function create(Magazine $magazine): array
     {
+        $markdownSummary = $magazine->description;
+
+        if (!empty($magazine->rules)) {
+            $markdownSummary .= "\r\n\r\n### Rules\r\n\r\n".$magazine->rules;
```

Comment:
my concern here is that this isn't localized, and of course it's difficult to know what language the magazine owner wanted

Reply:
You're right it is not. The "cleanest" way would be to introduce a `rules` property to the AP object, but that wouldn't be understood by lemmy, so it is kind of useless this way. I thought about completely removing the field, because, well it can't be displayed via AP...
We could use the language of the owner, but that is not that good eigher. We could introduce a language field in the magazine that defaults to english, but what good is that if it is only used for the rules... And how do we parse it back into our rules field...
It just complicates everything, but this way it at least federates somewhat :D

Reply:
I don't like it that much either, but it is understood by our software, so a user with a German interface will see it correctly under the "Regeln" header, because we put it in a different section. For users of other software it will always be a "Rules" header within the summary.

Reply:
What if it was just a `---` line instead? On our side it would say rules and outside would just be an ordered list? that doesn't seem to bad to me, but that's just my opinion

Reply:
It's ambigous. A lemmy magazine could have it just as a separator for whatever reason...

Reply:
we could maybe do a double line. That would look funny on other software though :thinking:

Reply:
Does anyone have a good idea how to handle this? Or is hardcoding "Rules" the best we have? :sweat_smile: 

I thought about it yesterday and I really don't want to remove that field, it could be really good for moderation improvements, like referencing a rules in a report and so on... So maybe we should introduce a magazine language field?

Reply:
My opinion is we should fully deprecate the rules field as I don't see it being useful considering it isn't part of the AP spec

Incoming remote group descriptions should stay as they are, kept as a description without us trying to regex/explode/parse them into our view of things, as we are messing with assuming how remotes parties expected their description to be read

The rules form text area should be hidden if no rules are present on the magazine, thus discouraging new magazines from setting any

Outgoing could be hardcoded to adding `Rules` for now and appending it to description. We could eventually add a notification that if you haven't moved rules into description yet it will be done automatically in a future release, at that point we migrate rules into the description column and drop it

Reply:
If we decide that the rules field does not provide enough value to keep it, I totally agree with the plan you have.

Here are my thoughts why the rules field could be very valuable:
If you are familiar with mastodon, then you know that each server has a set of rules which a user must agree to before they can register on a server. On mastodon these rules are not federated, as they are server specific. In the threadiverse however magazine rules are not server specific, so there should be a way to federate them. At the moment we do that by just adding them to the summary. So my vision is a mastodon server rule like implementation for magazines, where you 
- have a list of rules which are stored as an array
- can specify which rule was broken when reporting somebody
- have to agree to a magazine's rules before being able to post anything

This would of course need to be written down as an fep of sorts and it would be hard to keep it compatible with Lemmy if they don't implement it, but I think it is doable, even it it is harder and I think it would be valuable for moderation (though I don't know, I am not a moderator of a big community)